Docs/argocd-deployment-guide.md

```diff
@@ -0,0 +1,145 @@
+# ArgoCD Setup and Authentication Documentation
+
+1. ## Overview of the Project
+
+This project uses ArgoCD for managing Kubernetes applications with a secure authentication mechanism via Keycloak. The setup is fully automated using Terraform to deploy ArgoCD and configure the authentication via OIDC (OpenID Connect) with Keycloak as the identity provider.
+
+2. ## Prerequisites
+
+    Before deploying ArgoCD, ensure the following:
+
+    1. An AWS account, which you will need to deploy you cluster on AWS and with which resources like ACM certificates may be generated.
+
+
+    2. Please read and understand  project before proceeding <ins> ***[README](https://github.com/ADORSYS-GIS/webank-devops/blob/main/terraform/README.md)***</ins>, you can also read this too, try to understand it.. it will open your mind as we proceed <ins> ***[PROJECT ENV](https://github.com/ADORSYS-GIS/webank-devops/blob/main/Docs/deploy_workflow.md)***</ins>
+    
+3. ## Terraform Configuration for ArgoCD Deployment
+Our ArgoCD is deployed using Terraform. The deployment configuration is already included in the  <ins> ***[eks](https://github.com/ADORSYS-GIS/webank-devops/blob/main/terraform/eks.tf)***</ins> file, which ensures that ArgoCD is automatically installed and configured in the Kubernetes cluster.
+
+```hcl
+ enable_argocd = true
+  argocd = {
+    name          = "argocd"
+    chart_version = "6.7.3"
+    repository    = "https://argoproj.github.io/argo-helm"
+    namespace     = "argocd"
+    values = [
+      templatefile("${path.module}/files/argocd-values.yaml", {
+        domain                = local.argocdDomain,
+        name                  = local.name,
+        certArn               = var.cert_arn,
+        oidc_kc_client_id     = var.oidc_kc_client_id,
+        oidc_kc_client_secret = var.oidc_kc_client_secret,
+        oidc_kc_issuer_url    = var.oidc_kc_issuer_url,
+      })
+    ]
+  }
+```
+ This configuration:
+    1. enable_argocd = true: This part of the configuration controls whether ArgoCD will be deployed or not, for our case, yes
+    2. Enables ArgoCD installation with Helm.
+    3. Specifies the version of the ArgoCD chart (6.7.3).
```

Comment:
Specify the latest version of argo-cd's chart instead here

Reply:
okay sir, meaning i will have to also change the version in the `eks.tf` file right? because it made use of the version `6.7.3`